src/modules/authorizer/role/AUT_Roles_v2.sol

```diff
@@ -0,0 +1,449 @@
+    }
+
+    /// @notice Initializes the role authorizer.
+    /// @param  initialAdmin_ The initial admin of the role authorizer.
+    function __RoleAuthorizer_init(address initialAdmin_)
+        internal
+        onlyInitializing
+    {
+        if (initialAdmin_ == address(0)) {
+            revert Module__Authorizer__InvalidInitialAdmin();
+        }
+
+        // Start with 1 to account for the two native roles:
+        // DEFAULT_ADMIN_ROLE at 0 and PUBLIC_ROLE at 1.
+        _lastAssignedRoleId = 1;
+
+        // Note about DEFAULT_ADMIN_ROLE:
+        // The admin of the workflow holds the DEFAULT_ADMIN_ROLE, and has
+        // admin privileges on all modules in the contract.
+        // It is defined in the AccessControl contract and identified with
+        // bytes32("0x00").
+        // Modules can opt out of this on a per-role basis by setting the admin
+        // role to "BURN_ADMIN_ROLE".
+
+        // make the BURN_ADMIN_ROLE immutable.
+        _setRoleAdmin(BURN_ADMIN_ROLE, BURN_ADMIN_ROLE);
+
+        // set the initial admin as the DEFAULT_ADMIN_ROLE
+        _grantRole(DEFAULT_ADMIN_ROLE, initialAdmin_);
+    }
+
+    // ========================================================================
+    // Public Getter Functions
+
+    // ------------------------------------------------------------------------
+    // Getter -  Role Management
+
+    /// @inheritdoc IAuthorizer_v2
+    function getAdminRole() external pure returns (bytes32 defaultAdminId_) {
+        return DEFAULT_ADMIN_ROLE;
+    }
+
+    // ------------------------------------------------------------------------
+    // Getter -  Authorization
+
+    /// @inheritdoc IAuthorizer_v2
+    function getPermissions(address target_, bytes4 selector_)
+        external
+        view
+        virtual
+        returns (bytes32[] memory permissions_)
+    {
+        permissions_ = _permissions[target_][selector_];
+    }
+
+    /// @inheritdoc IAuthorizer_v2
+    function getLastAssignedRoleId()
+        external
+        view
+        returns (uint lastAssignedRoleId_)
+    {
+        lastAssignedRoleId_ = _lastAssignedRoleId;
+    }
+
+    /// @inheritdoc IAuthorizer_v2
+    function isRolePermissioned(
+        address target_,
+        bytes4 selector_,
+        bytes32 roleId_
+    ) public view virtual returns (bool isRolePermissioned_) {
+        bytes32[] memory permissions_ = _permissions[target_][selector_];
+        for (uint i = 0; i < permissions_.length; i++) {
+            if (permissions_[i] == roleId_) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /// @inheritdoc IAuthorizer_v2
+    function hasPermission(address caller_, address target_, bytes4 selector_)
+        external
+        view
+        virtual
+        returns (bool hasPermission_)
+    {
+        // If caller is the admin, they can call any function.
+        if (hasRole(DEFAULT_ADMIN_ROLE, caller_)) {
+            return true;
+        }
```

Comment:
Is there no way to restrict the default admin to calling a function? I could see usecases whereby a function should (from a legal perspective) be restricted to only a specific address which should for example could be held by an external party for regulation needs